### PR TITLE
Changes for Embarcadero C++ clang-based compilers, targeting Boost 1.74

### PR DESCRIPTION
**Pull request by eldiener**

### CODE REVIEW

**Comment by eldiener** [comment]
Can this please be merged ?

File: include/boost/smart_ptr/detail/shared_count.hpp [hunk 1]
```diff
@@ -661,7 +661,7 @@ inline shared_count::shared_count( weak_count const & r, sp_nothrow_tag ) BOOST_
 #pragma GCC diagnostic pop
 #endif
 
-#ifdef __BORLANDC__
+#ifdef BOOST_BORLANDC
```
Comment (pdimov) [inline include/boost/smart_ptr/detail/shared_count.hpp:664]: This change doesn't agree with the previous one; this uses BOOST_BORLANDC, the other does not.

Comment (eldiener) [reply include/boost/smart_ptr/detail/shared_count.hpp:664]: If config.hpp is not included or not yet included we can not use BOOST_BORLANDC but instead must use the long form of __BORLANDC__ and !__clang__.

Comment (pdimov) [reply include/boost/smart_ptr/detail/shared_count.hpp:664]: If you use one condition for the `# pragma warn -8027`, you must use the same condition for the subsequent `# pragma warn .8027`.

Comment (eldiener) [reply include/boost/smart_ptr/detail/shared_count.hpp]: Done !

**Comment by eldiener** [comment]
Is there any reason this can not be merged ? There are no conflicts and all tests pass.

**Pull request merged**